src/WindowsUtils/include/WindowsUtils/BusyLoopLauncher.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WINDOWS_UTILS_BUSY_LOOP_LAUNCHER_H_
+#define WINDOWS_UTILS_BUSY_LOOP_LAUNCHER_H_
+
+#include "OrbitBase/Result.h"
+#include "WindowsUtils/BusyLoopUtils.h"
+#include "WindowsUtils/Debugger.h"
+
+namespace orbit_windows_utils {
+
+// Utility to launch a process and install a busy loop at entry point. This is mainly used to allow
+// dll injection as early as possible during process creation. The class is single-use only, it can
+// not be reused to launch multiple processes.
```

Comment:
An easy way to enforce this is to make StartWithBusyLoopAtEntryPoint into a static factory method. Like this:

```
static std::unique_ptr<BusyLoopLauncher> Create(
      const std::filesystem::path& executable, const std::filesystem::path& working_directory,
      const std::string_view arguments);
```

The BusyLoopInfo would need to become a member then.

I think it is fine like this as well - feel free to ignore.

Reply:
static ErrorMessageOr<std::unique_ptr<BusyLoopLauncher>> Create(...) I guess. Yeah interesting idea, thanks. I think I prefer it as is for simplicity.